cogeo_mosaic/utils.py

```diff
@@ -394,95 +386,95 @@ def update_mosaic(
-    url_info = urlparse(url)
-
-    if url_info.scheme == "s3":
-        bucket = url_info.netloc
-        key = url_info.path.strip("/")
-        body = _aws_get_data(key, bucket)
-
-    elif url_info.scheme in ["http", "https"]:
-        # use requests
-        body = requests.get(url)
-        body = body.content
-
-    else:
-        with open(url, "rb") as f:
-            body = f.read()
-
-    if url.endswith(".gz"):
-        body = _decompress_gz(body)
-
-    if isinstance(body, dict):
-        return body
-    else:
-        return json.loads(body)
-
-
-@functools.lru_cache(maxsize=512)
-def fetch_mosaic_definition(url: str) -> Dict:
-    """Get Mosaic definition info."""
-    return get_mosaic_content(url)
-
-
-def fetch_and_find_assets(mosaic_path: str, x: int, y: int, z: int) -> Tuple[str]:
-    """Fetch mosaic definition file and find assets."""
-    mosaic_def = fetch_mosaic_definition(mosaic_path)
-    return get_assets(mosaic_def, x, y, z)
-
-
-def fetch_and_find_assets_point(mosaic_path: str, lng: float, lat: float) -> Tuple[str]:
-    """Fetch mosaic definition file and find assets."""
-    mosaic_def = fetch_mosaic_definition(mosaic_path)
-    min_zoom = mosaic_def["minzoom"]
-    quadkey_zoom = mosaic_def.get("quadkey_zoom", min_zoom)  # 0.0.2
-    tile = mercantile.tile(lng, lat, quadkey_zoom)
-
-    return get_assets(mosaic_def, tile.x, tile.y, tile.z)
-
-
-def get_assets(mosaic_definition: Dict, x: int, y: int, z: int) -> Tuple[str]:
-    """Find assets."""
-    min_zoom = mosaic_definition["minzoom"]
-
-    mercator_tile = mercantile.Tile(x=x, y=y, z=z)
-    quadkey_zoom = mosaic_definition.get("quadkey_zoom", min_zoom)  # 0.0.2
-
-    # get parent
-    if mercator_tile.z > quadkey_zoom:
-        depth = mercator_tile.z - quadkey_zoom
-        for i in range(depth):
-            mercator_tile = mercantile.parent(mercator_tile)
-        quadkey = [mercantile.quadkey(*mercator_tile)]
-
-    # get child
-    elif mercator_tile.z < quadkey_zoom:
-        depth = quadkey_zoom - mercator_tile.z
-        mercator_tiles = [mercator_tile]
-        for i in range(depth):
-            mercator_tiles = sum([mercantile.children(t) for t in mercator_tiles], [])
-
-        mercator_tiles = list(filter(lambda t: t.z == quadkey_zoom, mercator_tiles))
-        quadkey = [mercantile.quadkey(*tile) for tile in mercator_tiles]
-    else:
-        quadkey = [mercantile.quadkey(*mercator_tile)]
-
-    assets = list(
-        itertools.chain.from_iterable(
-            [mosaic_definition["tiles"].get(qk, []) for qk in quadkey]
-        )
-    )
-
-    # check if we have a mosaic in the url (.json/.gz)
-    return list(
-        itertools.chain.from_iterable(
-            [
-                fetch_and_find_assets(asset, x, y, z)
-                if os.path.splitext(asset)[1] in [".json", ".gz"]
-                else [asset]
-                for asset in assets
-            ]
-        )
-    )
+# def get_mosaic_content(url: str) -> Dict:
```

Comment:
@vincentsarago 
All this commented out code is now redundant, but removing it would presumably be a breaking change. Is a breaking change ok?

Reply:
breaking change is totally fine ;-)